lib/pliny/librato/metrics/backend.rb

```diff
@@ -7,35 +7,74 @@ module Metrics
       # Implements the Pliny::Metrics.backends API. Puts any metrics sent
       # from Pliny::Metrics onto a queue that gets submitted in batches.
       class Backend
-        attr_reader :queue
+        POISON_PILL = :'❨╯°□°❩╯︵┻━┻'
 
-        def initialize(source: nil, interval: 60, count: 1000, queue: nil)
-          @queue = queue || ::Librato::Metrics::Queue.new(
-            source:              source,
-            autosubmit_interval: interval,
-            autosubmit_count:    count
-          )
-          flush_on_shutdown
+        def initialize(source: nil, interval: 10, count: 500, **opts)
+          @metrics_queue = opts.fetch(:metrics_queue, Queue.new)
+          @librato_queue = opts.fetch(:librato_queue,
+                                      ::Librato::Metrics::Queue.new(
+                                        source:              source,
+                                        autosubmit_interval: interval,
+                                        autosubmit_count:    count
+                                      ))
         end
 
         def report_counts(counts)
-          report(counts)
+          metrics_queue.push(counts)
         end
 
         def report_measures(measures)
-          report(measures)
+          metrics_queue.push(measures)
+        end
+
+        def start
+          start_thread
+          Signal.trap('TERM') { stop }
```

Comment:
I think we're OK catching the `TERM` signal here, because we'll be calling this method in an initializer, and not in a thread.